### PR TITLE
Port Throw when UseAuthorization is incorrectly configured

### PR DESCRIPTION
**Pull request by pranavkm**
#### Description
Endpoint routing throws if an endpoint requires Auth or Cors middleware to execute but is not configured. https://github.com/aspnet/AspNetCore/issues/14049 pointed out that this does not throw if the Auth or Cors middleware are configured to execute before the endpoint routing middleware. In this case, an endpoint that requires authorization will run unauthenticated.

This change ports the runtime portion of https://github.com/aspnet/AspNetCore/pull/14401 to 3.0 to correctly flag this scenario.

#### Customer Impact
Endpoint routing will throw (result in a 500 response) if an endpoint requires authorization but the middleware is incorrectly configured.

#### Regression?
No. The middleware is new to 3.0.

#### Risk
Low. While the impact is very visible as an exception, this prevents an insecure configuration.

### CODE REVIEW

File: src/Http/Routing/test/FunctionalTests/EndpointRoutingIntegrationTerst.cs [hunk 1]
```diff
@@ -0,0 +1,240 @@
+// Copyright (c) .NET Foundation. All rights reserved.
```
Comment (Tratcher) [inline src/Http/Routing/test/FunctionalTests/EndpointRoutingIntegrationTerst.cs]: `IntegrationTerst.cs`

**Review by Tratcher — APPROVED** [review]

**Comment by leecow** [comment]
3.0.1 approved ... merge.

**Comment by pranavkm** [comment]
@aspnet/build could you merge this?

**Comment by mkArtakMSFT** [comment]
Thanks @pranavkm

**Pull request merged**